app/assets/javascripts/discourse/controllers/history.js.es6

```diff
@@ -29,6 +29,23 @@ export default Ember.Controller.extend(ModalFunctionality, {
     Discourse.Post.showRevision(postId, postVersion).then(() => this.refresh(postId, postVersion));
   },
 
+  revert(postId, postVersion) {
+    Discourse.Post.revertToRevision(postId, postVersion).then((result) => {
+      this.refresh(postId, postVersion);
+      if (result.topic) {
+        // TO-DO: update topic title here
```

Comment:
Also then you wouldn't need `Discourse.Post`, just a member function `post.revertToRevision(postVersion).then`...